synapse/storage/databases/main/metrics.py

```diff
@@ -320,7 +320,7 @@ def _get_start_of_day(self):
         """
         Returns millisecond unixtime for start of UTC day.
         """
-        now = time.gmtime()
+        now = time.gmtime(self.hs.get_clock().time())
```

Comment:
We try and avoid using `self.hs` directly where possible (it can hide problems that should be caught at start up until later).

```suggestion
        now = time.gmtime(self._clock.time())
```